pretrained_models/alexnet.py

```diff
@@ -0,0 +1,187 @@
+    for c in children:
+        depthfirstsearch(int(c), out)
+    return out
+
+
+def synset_to_dfs_ids(synset):
+    ids = [x for x in depthfirstsearch(synset) if x <= 1000]
+    ids = [corr[x] for x in ids]
+    return ids
+
+def AlexNet(weights_path=None, heatmap=False):
+    if heatmap:
+        inputs = Input(shape=(3, None, None))
+    else:
+        inputs = Input(shape=(3, 227, 227))
+
+    conv_1 = Convolution2D(96, 11, 11, subsample=(4, 4), activation='relu',
+                           name='conv_1')(inputs)
+
+    conv_2 = MaxPooling2D((3, 3), strides=(2, 2))(conv_1)
+    conv_2 = crosschannelnormalization(name='convpool_1')(conv_2)
+    conv_2 = ZeroPadding2D((2, 2))(conv_2)
+    conv_2 = merge([
+                       Convolution2D(128, 5, 5, activation='relu', name='conv_2_' + str(i + 1))(
+                           splittensor(ratio_split=2, id_split=i)(conv_2)
+                       ) for i in range(2)], mode='concat', concat_axis=1, name='conv_2')
+
+    conv_3 = MaxPooling2D((3, 3), strides=(2, 2))(conv_2)
+    conv_3 = crosschannelnormalization()(conv_3)
+    conv_3 = ZeroPadding2D((1, 1))(conv_3)
+    conv_3 = Convolution2D(384, 3, 3, activation='relu', name='conv_3')(conv_3)
+
+    conv_4 = ZeroPadding2D((1, 1))(conv_3)
+    conv_4 = merge([
+                       Convolution2D(192, 3, 3, activation='relu', name='conv_4_' + str(i + 1))(
+                           splittensor(ratio_split=2, id_split=i)(conv_4)
+                       ) for i in range(2)], mode='concat', concat_axis=1, name='conv_4')
+
+    conv_5 = ZeroPadding2D((1, 1))(conv_4)
+    conv_5 = merge([
+                       Convolution2D(128, 3, 3, activation='relu', name='conv_5_' + str(i + 1))(
+                           splittensor(ratio_split=2, id_split=i)(conv_5)
+                       ) for i in range(2)], mode='concat', concat_axis=1, name='conv_5')
+
+    dense_1 = MaxPooling2D((3, 3), strides=(2, 2), name='convpool_5')(conv_5)
+
+    if heatmap:
+        dense_1 = Convolution2D(4096, 6, 6, activation='relu', name='dense_1')(dense_1)
+        dense_2 = Convolution2D(4096, 1, 1, activation='relu', name='dense_2')(dense_1)
+        dense_3 = Convolution2D(1000, 1, 1, name='dense_3')(dense_2)
+        prediction = Softmax4D(axis=1, name='softmax')(dense_3)
+    else:
+        dense_1 = Flatten(name='flatten')(dense_1)
+        dense_1 = Dense(4096, activation='relu', name='dense_1')(dense_1)
+        dense_2 = Dropout(0.5)(dense_1)
+        dense_2 = Dense(4096, activation='relu', name='dense_2')(dense_2)
+        dense_3 = Dropout(0.5)(dense_2)
+        dense_3 = Dense(1000, name='dense_3')(dense_3)
+        prediction = Activation('softmax', name='softmax')(dense_3)
+
+    model = Model(input=inputs, output=prediction)
+
+    if weights_path:
+        model.load_weights(weights_path)
+
+    return model
+
+
+if __name__ == "__main__":
+    # Test pretrained model
+    model = AlexNet('alexnet_weights.h5')
+    model.summary()
```

Comment:
I think we should remove that, one could always load the model by importing it from pretrained models and then do the summary. We should not have `main` stuff in this library IMO.

Reply:
Ok, I will make the change!